### PR TITLE
Feature/2016 april update

### PR DESCRIPTION
**Pull request by nmwd**

### CODE REVIEW

File: css-sass.md [hunk 1]
```diff
@@ -184,37 +237,40 @@ It's not really important for your compiled CSS to be easy to read since only .s
...
 
-### Compass
+### Node Sass and Mixin Libraries
+
+We have moved away from using mixin libraries such as Compass or Bourbon and away from Ruby Sass as our compiler. We now use a combination of Node Sass, a few selected mixins and PostCSS (see below for more information) to compile our Sass files. This removes bloat from the compiled stylesheet and improves compile time drastically (over 10 seconds with Ruby vs 1 second with Node).
+
+If legacy projects still use either Compass or Ruby Sass, just use what is in place. There is no need to refactor the workflow.
 
-Use it. On every project. It's great.
+## PostCSS
```
Comment (nmwd) [inline css-sass.md:246]: @reubenmoes, is there anything more you want to add about PostCSS here?

File: css-sass.md [hunk 2]
```diff
@@ -400,38 +424,29 @@ Know that CSS comments are included in compiled CSS whereas [SASS style comments
...
+Breakpoints should be defined in _variables.scss under the $breakpoints map and can be called in scss files with the `breakpoint` function in _functions.scss. Breakpoints shouldn't be named after specific devices or classes of device. $breakpoints(iphone) is a really bad breakpoint name, as evident by there being 3 sizes of iPhone being sold at the time writing. More abstract names are encouraged.
 
 ### Don't build mobile first (sorry)
 
 Contrary to the cool kids, don't build your mobile layout first using media queries for desktop layouts. This creates a situation where you need to polyfill media query support for older browsers and end up with strange conflicts in things like harvey.js and enquire.js. This also creates unnecessary overhead. There's no need at all for IE8 to support media queries. You should build so your average sized desktop layout works without media queries, then use media queries to adapt smaller and larger, as the case may be. The only exception is in situations where you're only building for mobile, or only supporting browsers with media query support. Otherwise, take this approach.
 
+## Modules
```
Comment (nmwd) [inline css-sass.md]: Should we also talk about includes and vendors here?

File: css-sass.md [hunk 3]
```diff
@@ -400,38 +424,29 @@ Know that CSS comments are included in compiled CSS whereas [SASS style comments
...
 
 Buttons are a good example of a potential module, as are forms.
 
-### Partials
-
-Partials are pieces of code that aren't modular in nature, but are broken off for easier maintainability. Good examples of potential partials are headers, footers, and specifically designed un-reusable items.
-
-Each partial should be contained in its own file. You shouldn't have a file called `parials.scss`, rather each module should be individually named, prefixed with an underscore, in the partials directory. For example `/partials/_header.scss`.
-
 ## Including third party libraries and utilities
```
Comment (nmwd) [inline css-sass.md]: Should this be replaced with the vendors folder?

File: css-sass.md (hunk 1 above)
Comment (igorbarbashin) [inline css-sass.md:246]: I'd add a couple of examples how Autoprefixer can be used on a project.

E.g. gulp:

```
gulp.task('styles', function () {
    var sass = require('gulp-sass');
    var postcss = require('gulp-postcss');
    var sourcemaps = require('gulp-sourcemaps');
    var autoprefixer = require('autoprefixer');

    return gulp.src('./src/*.sass')
        .pipe(sourcemaps.init())
        .pipe(sass())
        .pipe(postcss([ autoprefixer({ browsers: ['last 5 versions'] }) ]))
        .pipe(sourcemaps.write('.'))
        .pipe(gulp.dest('./dest'));
});
```

Comment (nmwd) [inline css-sass.md:246]: I wonder if we should add another page/section on both gruntyplate and gulpyplate. The plan eventually is to get rid of Grunt right?

File: css-sass.md [hunk 4]
```diff
@@ -310,85 +361,58 @@ When nesting declaration blocks you should match the selector in a SASS style co
...
+
         &.secondary {
+
             .sidebar & {
-                
+
             }
+
         }
+
```
Comment (igorbarbashin) [inline css-sass.md:383]: I'm still not quite sold on this spacing between closing braces

File: css-sass.md [hunk 5]
```diff
@@ -457,54 +469,38 @@ Instead of this:
 
     .button {
         @include border-radius(3px);
```
Comment (igorbarbashin) [inline css-sass.md]: Compass mixin. Better be `border-radius: 3px`

File: css-sass.md (hunk 1 above)
Comment (igorbarbashin) [inline css-sass.md:246]: Eventually yes. I think for this PR we don't need gruntyplate and gulpyplate updates. But we should start outlining it

File: css-sass.md (hunk 4 above)
Comment (nmwd) [inline css-sass.md:383]: With a line break before and after?

File: css-sass.md (hunk 5 above)
Comment (nmwd) [inline css-sass.md]: Good catch, Will fix this.

File: css-sass.md (hunk 1 above)
Comment (reubenmoes) [inline css-sass.md:246]: What if we just listed the postcss plugins that we are fans of?  - autoprefixer, sourcemaps,

File: css-sass.md [hunk 6]
```diff
@@ -154,13 +159,61 @@ For example:
...
     .sidebar .widget, .site_footer .widget { border-bottom: 1px solid #333;padding-bottom: 1em;margin-bottom: 1em;}
 
-### Group like rules
-
-Group your CSS rules in similar chunks. For example, put everything to do with fonts together, as with box model, positioning, and so forth.
+### Rule order
+
+Style rules should be placed in the following order. Note that anything bringing in rules not declared presently are placed first so you know what you're bringing in.
+
+1.  includes
```
Comment (reubenmoes) [inline css-sass.md:166]: at first I wasn't sure what this meant.  Could we write `@include`s or `mixins`?

File: css-sass.md [hunk 7]
```diff
@@ -154,13 +159,61 @@ For example:
...
+2.  regular styles grouped by type:
+    1.  Positioning
+    2.  Display & Box Model
+    3.  Color
+    4.  Text
+    5.  Other
+
+Example:
+
+    .selector {
```
Comment (reubenmoes) [inline css-sass.md:176]: Does this need to be in a markdown code block?  ex ` `````` `

File: css-sass.md (hunk 2 above)
Comment (reubenmoes) [inline css-sass.md]: +1 ya

File: css-sass.md (hunk 7 above)
Comment (reubenmoes) [inline css-sass.md:176]: nevermind.  Seems to work fine :)

File: css-sass.md [hunk 8]
```diff
@@ -184,37 +237,40 @@ It's not really important for your compiled CSS to be easy to read since only .s
...
-* Variable names should observe the same underscore and hyphen pattern as with class names: underscores in place of spaces, hyphens to indicate a modifier.
-* Variable names should be all lowercase.
-* Variable names should focus on readability and specificity over saving space. For example, use `background` instead of `bkg`.
-* Variables should be named after what they are, not what they look like. For example, don’t have a variable called `$green`, but use something like `$color-brand` instead.
-* The word colour should be spelled the American way (color) for consistency with CSS, even if you're a Canadian that absolutely can't stand spelling it that way.
+*   Variable names should follow the [category]-[modifier] pattern. For example, all font variables should be prefixed with `$font-`, resulting in `$font-base`, `$font-body`, `$font-feature`, and so forth.
+*   Variable names should observe the same underscore and hyphen pattern as with class names: underscores in place of spaces, hyphens to indicate a modifier.
```
Comment (reubenmoes) [inline css-sass.md:280]: Is anybody is using underscores in place of spaces?

File: css-sass.md (hunk 6 above)
Comment (reubenmoes) [inline css-sass.md:166]: or maybe it's fine... I didn't realize it has the code block below

File: javascript.md [hunk 9]
```diff
@@ -1379,8 +80,6 @@ mainmenu: true
   - [innerHTML vs textContent for script text](http://jsperf.com/innerhtml-vs-textcontent-for-script-text)
   - [Long String Concatenation](http://jsperf.com/ya-string-concat)
 
-
-
 ## Resources
```
Comment (nmwd) [inline javascript.md]: Do we want to keep Resources as well as the Performance section? Are they still relevant?

Comment (reubenmoes) [inline javascript.md]: I think both can be removed.  They are in the airbnb guide
https://github.com/airbnb/javascript#performance

**Pull request merged**